extension/observer/k8sobserver/extension.go

```diff
@@ -27,6 +29,13 @@ var (
 	_ observer.Observable = (*k8sObserver)(nil)
 )
 
+var enableK8sObserverWatchErrorsMetric = featuregate.GlobalRegistry().MustRegister(
```

Comment:
I agree with @atoulme , there is no need for a feature gate here. Just ensuring this metrics is not enabled by default should be enough.